Document syncset. (#339)
openshift/hive: docs/syncidentityprovider.md

# SyncIdentityProvider

## Overview

> `SyncIdentityProvider` and `SelectorSyncIdentityProvider` manage cluster authentication for hive-managed clusters.

Reviewer: ```suggestion
Use `SyncIdentityProvider` and `SelectorSyncIdentityProvider` objects to manage cluster authentication for hive-managed clusters. You use these objects to share your identity providers among all valid clusters.

Like `SyncSets` and `SelectorSyncSets`, you use a `SyncIdentityProvider` to manage identity providers clusters that are in the same namespace as the `SyncIdentityProvider` object. To manage identity providers in clusters that match a specific tag, use a `SelectorSyncIdentityProvider`.
```

## SyncIdentityProvider Object Definition

> `SyncIdentityProvider` contains a list of `identityProviders` to patch in clusters referenced by `ClusterDeployment` name in the current namespace. A `SyncSet` will be generated for each `SyncIdentityProvider` and will contain a patch which modifies the cluster `oauth` object. 

Reviewer: ```suggestion
A `SyncIdentityProvider` contains a list of `identityProviders` to apply to specific clusters in that namespace. When you apply the `SyncIdentityProvider`, a `SyncSet` is generated for each `IdentityProvider` that contains a patch that modifies the cluster `oauth` object. 
```
